controllers/componentdetectionquery_controller.go

```diff
@@ -256,15 +256,27 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 			dockerfileContextMap[context] = updatedLink
 		}
 
+		for context, link := range devfileContextMap {
+			updatedLink, err := devfile.UpdateGitLink(source.URL, source.Revision, link)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to update the devfile link %v", req.NamespacedName))
+				r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, err)
+				return ctrl.Result{}, nil
+			}
+			devfilesURLMap[context] = updatedLink
+		}
+
 		for context := range devfilesMap {
```

Comment:
wondering if we even need this for block

Reply:
We need it for single component repos, I believe.